Introduce Deming/Orthogonal regression

JesperGr opened the pull request:
Deming/Orthogonal regression, least-Squares fitting the points in the 2D dataset (x,y) to a line
`
   a*x + b*y + c = 0
`
For delta equal 1 (the default value), this is performing orthogonal regression, minimizing the sum of squared perpendicular distances from the data points to the regression line.

Discussed/suggested here:
https://github.com/mathnet/mathnet-numerics/discussions/786

Questions: 
* Is `Fit.Line2D` a good name?
* Where to put `StandardLineToYxLine`?

jkalias commented:
> * Is `Fit.Line2D` a good name?
I think that `Fit.Deming` would be more appropriate. `Line2D` seems like it performs the "usual" least squares fitting

> * Where to put `StandardLineToYxLine`?
This is used purely for testing purposes, I think it is more appropriate in the testing class `FitTests.StandardLineToYxLine `

jkalias commented on src/Numerics/LinearRegression/DemingRegression.cs:
```diff
@@ -0,0 +1,203 @@
+
+            if (x.Length <= 1)
+            {
+                throw new ArgumentException($"A regression of the requested order requires at least {2} samples. Only {x.Length} samples have been provided.");
```
2 doesn't need to be in braces {}

JesperGr replied:
Fixed also in SimpleRegression.cs

jkalias commented on src/Numerics/LinearRegression/DemingRegression.cs:
```diff
@@ -0,0 +1,203 @@
+        /// <param name="delta">Ratio of variances of x and y data, var(y)/var(x). Default value is 1</param>
+        /// <returns> returning its best fitting parameters as (a, b, c) tuple.</returns>
+        /// <remarks>
+        /// Solution algorithm as described from:
```
I couldn't find the reference, only a wikipedia [linke](https://en.wikipedia.org/wiki/Deming_regression). Could you provide a pdf link or something similar?

JesperGr replied:
That was the best link I could find.

jkalias commented on src/Numerics/LinearRegression/DemingRegression.cs:
```diff
@@ -0,0 +1,203 @@
+                    return (1, 0, -mx);
+                // No line could be calculated
+                else
+                    return (0, 0, 0);
```
I would expect an algorithm which fails to return `double.NaN`s instead of zeros

JesperGr commented:
I like the Line in the name, because it is very similar to the `SimpleRegression`. But I am fine with `Deming` as well, so I have updated the `Fit.Line2D` to `Fit.Deming`. And moved the `StandardLineToYxLine` to the `FitTests` - I will then just keep my own line tools alive.
